### DISCUSSION TITLE
How is SSG doing?

### DISCUSSION THREAD

**Discussion by rmortes** [solidjs/solid#686] [Q&A]
I've being looking for a way to statically generate my Solid landing page into individual html files, instead of redirecting everything to the index and letting the client render the page.

Just wanted to know how is the SSG doing, and if I can be of any help. Others tools can make do for now, but I know I'll have to add heavy reactivity down the line and I want to use Solid for that.

By the way, the current static site generator refuses to work with any typescript template. The typings are wrong (only acceps one route, not a list of them!) and even I rewrite it in plain JS it doesn't know that to do with .ts and .tsx files. Any idea on how could I fix that? Thanks!

**Comment by bobaekang** [comment]
It seems to me that your use case might be able to take advantage of [Astro](https://astro.build/) with Solid components--especially since [`solid-start`](https://github.com/solidjs/solid-start) is still far from being stable. In fact, @ryansolid recently did [a streaming on YouTube](https://www.youtube.com/watch?v=UPhx3OxUH0w) on just that--using Astro with Solid. Astro should take care of the static bits and Solid, the dynamic parts. Sharing state across Solid components is also apparently possible (watch [this part of the video](https://www.youtube.com/watch?v=UPhx3OxUH0w&t=1h15m56s)).

**Comment by ryansolid** [comment]
I can see it possible there are type issues. If you are talking about the SSR utility in `solid-ssr` it has the expectation that the server stuff has already been compiled. It's basically just the rendering part, runs after build, so Solid JSX should already be transformed and Typescript gone. I use the utility in small demos and in Solid Start's static adapter but I can't say the latter is really fleshed out. But the utility is just a provide a URL and render sort of deal.

I second that Astro is probably a pretty good choice if MPAs work for you.

**Comment by unoexperto** [comment]
@ryansolid Hey Ryan. Sorry for reviving the ancient thread. Are there updates on server-side generation since 2021 ? Thank you.

**Reply by znycheporuk** [reply]
https://docs.solidjs.com/solid-start/building-your-application/route-prerendering
